### PR TITLE
Feature/authentication

### PR DESCRIPTION
**Pull request by Merteg**

### CODE REVIEW

**Comment by Merteg** [comment]
Wait for your review

File: client/src/components/Home.jsx [hunk 1]
```diff
@@ -0,0 +1,11 @@
+import React from 'react';
+import { Row, Col } from 'react-materialize'
+
+
+export function Home() {
```
Comment (Merteg) [inline client/src/components/Home.jsx]: Better to use default export in end of the file

File: client/src/components/LogOut.jsx [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+      }
+      else {
+        this.setState({ msg: data.msg })
+      }
+      window.Materialize.toast(this.state.msg, 3000)
+    })
+  }
+
+  render() {
+    return <NavItem onClick={this.log_out} href="log-out">Log Out</NavItem>
```
Comment (Merteg) [inline client/src/components/LogOut.jsx]: I'm not sure that I really need a `href` attribute.

File: client/public/index.css [hunk 3]
```diff
@@ -0,0 +1,13 @@
+.nav-wrapper {
+    padding: 0 15px;
+}
+.input-field label{
```
Comment (serge30) [inline client/public/index.css]: As I understand you overrides this classes for input labels which affects all forms on the application, but those changes only need for login form on the header. In this case you need restrict this only for class of header.

File: client/src/components/Nav.jsx [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+
+
+export class Nav extends Component {
+  constructor(props) {
+    super(props);
+    this.update = this.update.bind(this);
+  }
+
+  isLogged() {
+    return typeof (sessionStorage['Authorization-token']) !== 'undefined' ? true : false
```
Comment (serge30) [inline client/src/components/Nav.jsx]: `typeof (sessionStorage['Authorization-token']) !== 'undefined'` already returns true or false, you don't need here ternary    operator

**Review by serge30 — CHANGES_REQUESTED** [review]

File: client/src/components/Nav.jsx (hunk 4 above)
Comment (Merteg) [reply client/src/components/Nav.jsx]: Also I think move this function into scripts.js file, is it ok?

Comment (serge30) [reply client/src/components/Nav.jsx]: no objections from my side

**Comment by Merteg** [comment]
I made a change

File: client/src/scripts.js [hunk 5]
```diff
@@ -0,0 +1,26 @@
+import { server_url } from './config.json'
```
Comment (serge30) [inline client/src/scripts.js]: I'm not sure `scripts.js` is a good name for this module, something like `utils.js` sounds better.

**Review by serge30 — CHANGES_REQUESTED** [review]

**Comment by Merteg** [comment]
Add new commit

**Review by serge30 — APPROVED** [review]
Looks good to me. But let's give a chance to team to review and provide feedback.

**Review by RayMet — APPROVED** [review]

**Review by ykobuk — APPROVED** [review]

**Review by yazhyvyi — APPROVED** [review]

**Pull request merged**